### PR TITLE
Web27 Display Filename

### PR DESCRIPTION
**Pull request by winnteas**
Why the changes are required

- to see what file is being edited on the editor page

Changes

- passed in filename when navigating to the editor page (via edit button or double clicking)
- filename shows on the left side of the header, buttons remain on the right side

image

### CODE REVIEW

**Review by james-teng-0 — DISMISSED** [review]
lgtm but please fix the frontend_testing pipeline before merging  - Error: Uncaught [Error: useLocation() may be used only in the context of a Router component.]

- you probably need to wrap the parent element in the Router component.

**Comment by james-teng-0** [comment]
I think you can use a parent useState in Dashboard.tsx and pass it to the Renderer and Sidebar so that you can pass the name to the Edit button. 
Similar to how the selectedFile / setSelectedFile is implemented.

File: frontend/src/deprecated/components/Editor/EditorHeader.tsx
Comment (lauraw0) [inline frontend/src/deprecated/components/Editor/EditorHeader.tsx:7]: pls remove unused imports!

File: frontend/src/packages/dashboard/Dashboard.tsx
Comment (lauraw0) [inline frontend/src/packages/dashboard/Dashboard.tsx:2]: useSelector is not used here so it can be removed :>

**Review by lauraw0 — APPROVED** [review]
just some style changes needed but besides that lgtm! you can merge once you have removed those unused imports

File: frontend/src/packages/dashboard/components/SideBar/SideBar.tsx
Comment (lauraw0) [inline frontend/src/packages/dashboard/components/SideBar/SideBar.tsx:124]: You can replace this code with the existing getFolderState() function in `/frontend/src/packages/dashboard/api/helpers.ts` !

**Pull request merged**
